src/src/com/microsoft/aad/adal/DateTimeAdapter.java

```diff
@@ -50,12 +50,31 @@
             DateFormat.DEFAULT);
 
     private final DateFormat iso8601Format = buildIso8601Format();
```

Comment:
This also bothers me. I can see why it happens in broker, broker always serialize datetime using default dateformat, different locale may cause the problem. But for DefaultTokenCache, the dateTimeAdapter is already introduced since 1.1.4. The only reason I can see why it happens is, user is only an old version of i.e. one drive, so they have existing tokens with old format. Then they update to an newer version, we start seeing the failures, does it sound reasonable?